yb-voyager/cmd/analyzeSchema.go

```diff
@@ -995,21 +996,30 @@ func reportDeferrableConstraintCreateTable(createTableNode *pg_query.Node_Create
 		*/
 		if column.GetColumnDef() != nil {
 			constraints := column.GetColumnDef().GetConstraints()
+			colName := column.GetColumnDef().GetColname()
 			if constraints != nil {
 				isForeignConstraint := false
 				isDeferrable := false
-				for _, constraint := range constraints {
+				var constraintType pg_query.ConstrType
+				for idx, constraint := range constraints {
 					if constraint.GetConstraint().Contype == pg_query.ConstrType_CONSTR_FOREIGN {
 						isForeignConstraint = true
 					} else if slices.Contains(deferrableConstraintsList, constraint.GetConstraint().Contype) {
+						//Getting the constraint's suffix before the DEFERRABLE clause as the clause is applicable to that constraint
+						if idx > 0 {
```

Comment:
Why do we need to check idx > 0 ? 
also comment above seems not relevant anymore? can you update/remove?

Reply:
Updated the comment.

> Why do we need to check idx > 0 ?

just to handle the case where `DEFERRABLE` clause is at the beginning of constraints, this is not a valid case but in case someone modifies DDL or any manual changes. So to handle the `index out of bound` error,  as we are using `idx-1`.